utils/terrascript_client.py

```diff
@@ -0,0 +1,325 @@
+        tss = {}
+        for name, config in self.configs.items():
+            ts = Terrascript()
+            ts += provider('aws',
+                           access_key=config['aws_access_key_id'],
+                           secret_key=config['aws_secret_access_key'],
+                           version=config['aws_provider_version'],
+                           region=config['region'])
+            b = backend("s3",
+                        access_key=config['aws_access_key_id'],
+                        secret_key=config['aws_secret_access_key'],
+                        bucket=config['bucket'],
+                        key=config['key'],
+                        region=config['region'])
+            ts += terraform(backend=b)
+            tss[name] = ts
+        self.tss = tss
+
+    def populate_configs_and_vars_from_vault(self):
+        self.init_accounts()
+        self.configs = self.get_vault_tf_secrets('config')
+        self.variables = self.get_vault_tf_secrets('variables')
+
+    def init_accounts(self):
+        config = get_config()
+        accounts = config['terraform']
+        self.accounts = accounts.items()
+
+    def get_vault_tf_secrets(self, type):
+        secrets = {}
+        for name, data in self.accounts:
+            secrets_path = data['secrets_path']
+            secret = vault_client.read_all(secrets_path + '/' + type)
+            secrets[name] = secret
+        return secrets
+
+    def populate(self, tf_query):
+        for namespace_info in tf_query:
+            # Skip if namespace has no terraformResources
+            tf_resources = namespace_info.get('terraformResources')
+            if not tf_resources:
+                continue
+            for resource in tf_resources:
+                self.populate_tf_resources(resource, namespace_info)
+        self.validate()
+
+    def populate_tf_resources(self, resource, namespace_info):
+        provider = resource['provider']
+        if provider == 'rds':
+            self.populate_tf_resource_rds(resource, namespace_info)
+        elif provider == 's3':
+            self.populate_tf_resource_s3(resource, namespace_info)
+        else:
+            raise UnknownProviderError(provider)
+
+    def populate_tf_resource_rds(self, resource, namespace_info):
+        account, identifier, values, output_resource_name = \
+            self.init_values(resource, namespace_info)
+
+        tf_resources = []
+        self.init_common_outputs(tf_resources, namespace_info,
+                                 output_resource_name)
+
+        # rds instance
+        variables = self.variables[account]
+        values['db_subnet_group_name'] = variables['rds-subnet-group']
+        values['vpc_security_group_ids'] = \
+            variables['rds-security-groups'].split(',')
+        values['password'] = \
+            self.determine_rds_db_password(namespace_info,
+                                           output_resource_name)
+
+        # rds outputs
```

Comment:
can you add some docs in here? this is hard to understand. I understand this is very specific to terrascript, but a small paragraph on what terrascript expects and what we are adding would come in handy

Reply:
sure